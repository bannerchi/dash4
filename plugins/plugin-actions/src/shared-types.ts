export type TButtonVariant =
	| 'primary'
	| 'secondary'
	| 'success'
	| 'danger'
	| 'warning'
	| 'info'
	| 'dark'
	| 'light'
	| 'link'
	| 'outline-primary'
	| 'outline-secondary'
	| 'outline-success'
	| 'outline-danger'
	| 'outline-warning'
	| 'outline-info'
	| 'outline-dark'
	| 'outline-light';

export interface IActionLink {
	type: 'link';
	title: string;
	href: string;
	image?: string;
	icon?: string;
}
export interface IActionLinkWithId extends IActionLink {
	id: string;
}

export interface IActionTeaserLink {
	title: string;
	href: string;
	buttonVariant?: TButtonVariant;
}
export interface IActionTeaser {
	type: 'teaser';
	title: string;
	subtitle?: string;
	link?: IActionLink | IActionLink[];
	image?: string;
	icon?: string;
}
export interface IActionTeaserWithId extends IActionTeaser {
	id: string;
}

export interface IClientConfig {
	title?: string;
	actions?: (IActionLinkWithId | IActionTeaserWithId)[];
}
