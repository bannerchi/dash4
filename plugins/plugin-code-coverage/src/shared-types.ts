export interface IThreshold {
	branches?: [number, number];
	functions?: [number, number];
	lines?: [number, number];
	statements?: [number, number];
}

export interface IThresholdGuaranteed {
	branches: [number, number];
	functions: [number, number];
	lines: [number, number];
	statements: [number, number];
}

export interface ICoverageSection {
	coverage: number;
	name: string;
	counter: number;
	covered: number;
}

export interface ICoverage {
	error?: boolean;
	message?: string;
	statements?: ICoverageSection;
	branches?: ICoverageSection;
	functions?: ICoverageSection;
	lines?: ICoverageSection;
}

export interface IClientConfig {
	cwd: string;
	dark?: boolean;
	threshold: IThresholdGuaranteed;
}
