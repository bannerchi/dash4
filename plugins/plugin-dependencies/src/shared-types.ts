export interface IInstallProcess {
	// custom title for installation process (default=npm run install)
	title?: string;
	// current working directory of the child process. (default = process.cwd())
	cwd?: string;
	// command which should be executed (default = 'npm run install')
	cmd?: string;
}

export interface IInstallProcessPrepared {
	id: string;
	title: string;
	cwd: string;
	cmd: string;
}

export interface IClientConfig {
	title?: string;
	cwd: string;
	installProcess: IInstallProcessPrepared;
}

export interface IPackageJson {
	[key: string]: any;
	name?: string;
	version?: string;
	description?: string;
	dependencies?: { [key: string]: string };
	devDependencies?: { [key: string]: string };
}

export interface IError {
	error: boolean;
	message: string;
}

export interface IDependency {
	version: string;
	latestVersion?: string;
	isUpToDate: boolean | null;
	pathes: string[];
	packages: string[];
}

export interface IDependencyObj {
	[key: string]: IDependency;
}

export interface IDependencyData {
	dependencies: IDependencyObj;
	devDependencies: IDependencyObj;
}

export const SEND_TO_SERVER_EVENT_NAMES = {
	connected: 'connected',
	update: 'update',
	install: 'install',
} as const;

export type ISendToServerEventNames = keyof typeof SEND_TO_SERVER_EVENT_NAMES;
export type ISendToServerData = void | string | void;
export type ISendToServer = (eventName: ISendToServerEventNames, data: ISendToServerData) => void | Promise<void>;

export type IRecieveFromClientEventNames = ISendToServerEventNames;
export type IRecieveFromClientCb = (data: ISendToServerData) => void | Promise<void>;
export type IRecieveFromClient = (
	eventName: IRecieveFromClientEventNames,
	cb: IRecieveFromClientCb
) => void | Promise<void>;

export const SEND_TO_CLIENT_EVENT_NAMES = {
	data: 'data',
} as const;

export type ISendToClientEventNames = keyof typeof SEND_TO_CLIENT_EVENT_NAMES;
export type ISendToClientData = IDependencyObj | IError;
export type ISendToClient = (eventName: ISendToClientEventNames, data: ISendToClientData) => void | Promise<void>;

export type IRecieveFromServerEventNames = ISendToClientEventNames;
export type IRecieveFromServerCb = (data: ISendToClientData) => void | Promise<void>;
export type IRecieveFromServer = (
	eventName: IRecieveFromServerEventNames,
	cb: IRecieveFromServerCb
) => void | Promise<void>;
