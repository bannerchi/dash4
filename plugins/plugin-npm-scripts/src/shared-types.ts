export type TButtonVariant =
	| 'primary'
	| 'secondary'
	| 'success'
	| 'danger'
	| 'warning'
	| 'info'
	| 'dark'
	| 'light'
	| 'link'
	| 'outline-primary'
	| 'outline-secondary'
	| 'outline-success'
	| 'outline-danger'
	| 'outline-warning'
	| 'outline-info'
	| 'outline-dark'
	| 'outline-light';

export interface IScript {
	title?: string;
	cmd: string;
	cwd?: string;
	buttonVariant?: TButtonVariant;
}

export interface IScriptWithId extends IScript {
	id: string;
	cwd: string;
}

export interface IClientConfig {
	scripts: IScriptWithId[];
}
