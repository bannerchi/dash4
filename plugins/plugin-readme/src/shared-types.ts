export interface IReadmeClientConfig {
	file: string;
	height?: number | string;
}

export interface IReadmeListItemClientConfig {
	title: string;
	file: string;
	id: string;
}

export interface IReadmeListClientConfig {
	title?: string;
	files: IReadmeListItemClientConfig[];
}
