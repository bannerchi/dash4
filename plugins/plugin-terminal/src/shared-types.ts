export interface IClientConfig {
	cmd: string;
	cwd?: string;
}
