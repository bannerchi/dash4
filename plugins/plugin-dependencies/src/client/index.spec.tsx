/// <reference types="@types/jest" />
import React from 'react';
import { fireEvent, render } from 'react-testing-library';
import { ISendToServer } from '../shared-types';

jest.mock('@dash4/plugin-npm-scripts/build/client/components/Script', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	NpmScript: () => <div />,
}));
const useDataMock = jest.fn();
jest.mock('./hooks', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	useData: useDataMock,
}));
jest.mock('@dash4/log/build/browser', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	error: jest.fn(),
	warn: jest.fn(),
	success: jest.fn(),
}));
const registerPluginMock = jest.fn();
jest.mock('@dash4/client/build/register-plugin', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	registerPlugin: registerPluginMock,
}));

import { PluginDependencies } from './index';

const wait = (duration: number = 100) => new Promise((resolve) => setTimeout(resolve, duration));

const exampleData = {
	react: {
		version: '16.8.4',
		latestVersion: '16.8.4',
		isUpToDate: true,
		pathes: [],
		packages: [],
	},
	'react-dom': {
		version: '16.8.4',
		latestVersion: '16.8.4',
		isUpToDate: true,
		pathes: [],
		packages: [],
	},
	'@types/react': {
		version: '16.8.3',
		latestVersion: '16.8.4',
		isUpToDate: false,
		pathes: [],
		packages: [],
	},
	'@types/node': {
		version: '10.0.0',
		latestVersion: '10.0.1',
		isUpToDate: false,
		pathes: [],
		packages: [],
	},
};

const props = {
	id: '1',
	name: 'PluginDependencies',
	lowerCaseName: 'plugin-dependencies',
	dark: false,
	clientConfig: {
		cwd: './',
		installProcess: {
			id: '1',
			title: 'install',
			cwd: './',
			cmd: 'npm run install',
		},
	},
};

// if ((global as any).document) {
// 	// @ts-ignore
// 	document.createRange = () => ({
// 		setStart: () => undefined,
// 		setEnd: () => undefined,
// 		// @ts-ignore
// 		commonAncestorContainer: {
// 			nodeName: 'BODY',
// 			ownerDocument: document,
// 		},
// 	});
// }

describe('PluginTerminal', () => {
	test('exist', () => {
		expect(typeof PluginDependencies).toBe('object');
	});
	test('render loading', () => {
		useDataMock.mockImplementation((id: string, sendFactory: (send: ISendToServer) => void) => {
			sendFactory(() => undefined);
			return;
		});
		const { container } = render(<PluginDependencies {...props} />);
		expect(container.firstChild).toMatchSnapshot();
		expect(registerPluginMock).toHaveBeenCalledWith('PluginDependencies', PluginDependencies);
	});
	test('render error', () => {
		useDataMock.mockImplementation((id: string, sendFactory: (send: ISendToServer) => void) => {
			sendFactory(() => undefined);
			return {
				error: true,
				message: 'error',
			};
		});
		const { container } = render(<PluginDependencies {...props} />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with data', async () => {
		useDataMock.mockImplementation((id: string, sendFactory: (send: ISendToServer) => void) => {
			sendFactory(() => undefined);
			return exampleData;
		});
		const { container } = render(<PluginDependencies {...props} />);
		expect(container.firstChild).toMatchSnapshot();
		// fireEvent.click(getByTestId('1_filter-button'));
		// await wait();
		// const inputFiled = getByPlaceholderText('search') as HTMLInputElement;
		// inputFiled.value = 'react';
		// await wait();
		// expect(container.firstChild).toMatchSnapshot();
	});
});
