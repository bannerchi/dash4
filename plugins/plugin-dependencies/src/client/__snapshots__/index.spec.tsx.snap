// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginTerminal render loading 1`] = `
<div
  class="Component-window-0-1-11 Component-windowBright-0-1-13 Component-windowAutoStretch-0-1-12"
  tabindex="1"
>
  <div
    class=" Component-header-0-1-16"
  >
    <p
      class="Component-title-0-1-18"
    >
      Dependencies
    </p>
    <div
      class="Component-button-0-1-17"
    >
      <button
        class="btn btn-outline-primary btn-sm"
        data-testid="1_filter-button"
        type="button"
      >
        <span
          class=" Component-root-0-1-21 Component-size-m-0-1-23"
        >
          <i
            class="material-icons Component-size-i-m-0-1-26"
          >
            filter_list
          </i>
        </span>
      </button>
    </div>
  </div>
  <div
    class="Component-windowBody-0-1-1  Component-body-0-1-35"
  >
    <div
      class="Component-spinner-0-1-3"
    >
      <div
        class="spinner-grow"
      />
    </div>
  </div>
</div>
`;

exports[`PluginTerminal render with data 1`] = `
<div
  class="Component-window-0-1-11 Component-windowBright-0-1-13 Component-windowAutoStretch-0-1-12"
  tabindex="1"
>
  <div
    class=" Component-header-0-1-16"
  >
    <p
      class="Component-title-0-1-18"
    >
      Dependencies
    </p>
    <div
      class="Component-button-0-1-17"
    >
      <button
        class="btn btn-outline-primary btn-sm"
        data-testid="1_filter-button"
        type="button"
      >
        <span
          class=" Component-root-0-1-21 Component-size-m-0-1-23"
        >
          <i
            class="material-icons Component-size-i-m-0-1-26"
          >
            filter_list
          </i>
        </span>
      </button>
    </div>
  </div>
  <div
    class="Component-windowBody-0-1-1  Component-body-0-1-35"
  >
    <ul
      class="Component-ul-0-1-6 "
    >
      <li
        class="DependencyRaw-li-0-1-36"
      >
        <div
          class="DependencyRaw-name-0-1-37"
        >
          react
        </div>
        <div
          class="DependencyRaw-version-0-1-38"
        >
          <span
            class="DependencyRaw-versionInCaseUpToDate-0-1-40"
          >
            16.8.4
          </span>
        </div>
      </li>
      <li
        class="DependencyRaw-li-0-1-36"
      >
        <div
          class="DependencyRaw-name-0-1-37"
        >
          react-dom
        </div>
        <div
          class="DependencyRaw-version-0-1-38"
        >
          <span
            class="DependencyRaw-versionInCaseUpToDate-0-1-40"
          >
            16.8.4
          </span>
        </div>
      </li>
      <li
        class="DependencyRaw-li-0-1-36"
      >
        <div
          class="DependencyRaw-name-0-1-37"
        >
          @types/react
        </div>
        <div
          class="DependencyRaw-version-0-1-38"
        >
          <span
            class="DependencyRaw-versionInCaseOutOfDate-0-1-41"
          >
            16.8.3
          </span>
          <button
            class="DependencyRaw-button-0-1-39 btn btn-danger btn-sm"
            type="button"
          >
            <span
              class=" Component-root-0-1-21 Component-size-m-0-1-23"
            >
              <i
                class="material-icons Component-size-i-m-0-1-26"
              >
                arrow_upward
              </i>
            </span>
          </button>
        </div>
      </li>
      <li
        class="DependencyRaw-li-0-1-36"
      >
        <div
          class="DependencyRaw-name-0-1-37"
        >
          @types/node
        </div>
        <div
          class="DependencyRaw-version-0-1-38"
        >
          <span
            class="DependencyRaw-versionInCaseOutOfDate-0-1-41"
          >
            10.0.0
          </span>
          <button
            class="DependencyRaw-button-0-1-39 btn btn-danger btn-sm"
            type="button"
          >
            <span
              class=" Component-root-0-1-21 Component-size-m-0-1-23"
            >
              <i
                class="material-icons Component-size-i-m-0-1-26"
              >
                arrow_upward
              </i>
            </span>
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
